integration-libs/opps/root/coupon-codes/opps-coupon-codes.module.ts

```diff
@@ -0,0 +1,32 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { NgModule } from '@angular/core';
+import { LOCATION_INITIALIZED_MULTI } from '@spartacus/core';
+import { oppsCouponCodesInterceptors } from './http-interceptors';
+import { OppsCouponCodesService } from './opps-coupon-codes.service';
+
+export function saveCouponCodesFactory(
+  service: OppsCouponCodesService
+): () => void {
+  return () => {
+    service.saveUrlCouponCodes();
+  };
+}
+
+@NgModule({
+  providers: [
+    ...oppsCouponCodesInterceptors,
+    OppsCouponCodesService,
```

Comment:
nitpick: it doesn't have to be provided in a module, becasue it's an eagerly loaded module.
It suffices to make it `@Injectable({providedIn: 'root'})`

Reply:
done